### PR TITLE
#NGG-227 eslint support

### PR DESCRIPTION
**Pull request by mastertheblaster**

### CODE REVIEW

File: grunt-sections/codestyle.js [hunk 1]
```diff
@@ -46,6 +53,31 @@ module.exports = function (grunt) {
...
+      },
+      test: {
+        options: {
+          jshintrc: featureDetector.isTestInAppFolderEnabled() ? 'app/test/.eslintrc' : 'test/.eslintrc'
+        },
+        files: [{
+          src: ['test/{spec,mock,e2e}/**/*.js', 'app/modules/**/*.test.js', 'app/test/{spec,mock,e2e}/**/*.js']
+        }]
+      }
+    },
```
Comment (shahata) [inline grunt-sections/codestyle.js]: can't this configuration be simplified to just look for the first `.eslintrc` in my directory or its parent directories?

File: grunt-sections/codestyle.js [hunk 2]
```diff
@@ -46,6 +53,31 @@ module.exports = function (grunt) {
...
+            'app/{scripts,modules}/**/*.js',
+            '!app/modules/**/*.test.js',
+            '!app/test/**/*.js',
+            '!app/scripts/lib/**/*.js'
+          ]
+        }]
+      },
+      test: {
+        options: {
+          jshintrc: featureDetector.isTestInAppFolderEnabled() ? 'app/test/.eslintrc' : 'test/.eslintrc'
```
Comment (shahata) [inline grunt-sections/codestyle.js]: `jshintrc` doesn't make sense here. I'm guessing that this config will work even if you just pass all of the file globs in one simple config and we don't really need `configFile` or two entries

**Comment by shahata** [comment]
@mastertheblaster I'd like us to use the following strategy:
1) change wix-gruntfile so that it will run eslint if project has a .eslintrc file
2) provide simple utility to move to eslint, but do not simply convert existing config. instead create a .eslintrc that simply says `{"extends": "wix/angular"}` (see https://github.com/wix-private/eslint-config-wix) - this means that once ppl run this utility they'll need to fix all errors.
3) at some point we will run this utility automatically on projects, and they will be forced to fix problems and move to eslint

**Comment by mastertheblaster** [comment]
@shahata
1) Updated wix-gruntfile to run only eslint (no .eslintrc generation).
2) Created PR inside wix-angular generator to have a simple utility for .eslintrc creation.

**Comment by mastertheblaster** [comment]
Found an issue closing for now.

**Pull request closed without merging**